### ISSUE TITLE
[java] GuardLogStatement: Should not need to guard parameterized log messages where the replacement arg is an array element

### ISSUE THREAD

**Issue by jbisotti** [pmd/pmd#5153]
**Affects PMD Version:** 
7.0.0

**Rule:**
[GuardLogStatement](https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#guardlogstatement)

**Description:**
I do not believe an argument, which is an array element, should result in a violation of this rule.

**Code Sample demonstrating the issue:**

```java
        final String[] foo = ...
        LOG.info(
                "Some message here : foo0={}",
                foo[0]
        );
```

**Expected outcome:**

PMD reports "... Rule:GuardLogStatement Priority:2 Logger calls should be surrounded by log level guards..".
I think that's a false positive.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
Run via maven-pmd-plugin (3.24.0)

**Comment by jsotuyod** [comment]
This is a little more nuanced, as not all array accesses are ok…

```java
        LOG.info(
            "Some message here : foo={}",
            foo()[0] // this is an array access… but one level deeper the AST it's actually a method call, and absolutely needs a guard
        );
```

#5169 should cover both scenarios ok
